specs/eventing/control-plane.md

```diff
@@ -0,0 +1,1120 @@
+
+```yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: knative-developer
+rules:
+  - apiGroups: ["eventing.knative.dev"]
+    resources: ["broker", "trigger"]
+    verbs: ["get", "list", "create", "update", "patch", "delete"]
+  - apiGroups: ["messaging.knative.dev"]
+    resources: ["channel", "subscription"]
+    verbs: ["get", "list", "create", "update", "patch", "delete"]
+```
+
+In order to support resolving resources which meet the
+[Addressable](./overview.md#addressable) contract, the system controlling the
+[Trigger](#trigger-lifecycle) or [Subscription](#subscription-lifecycle) will
+need _read_ access to these resources. On Kubernetes, this is most easily
+achieved using role aggregation; on systems using Kubernetes RBAC, resources
+which wish to participate in Addressable resolution should provide the following
+`ClusterRole`:
+
+```yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: <component>-addressable-resolver
+  labels:
+    duck.knative.dev/addressable: "true"
+rules:
+  - apiGroups: [<resource apiGroup>]
+    resources: [<resource kind >, <resource kind>/status]
+    verbs: ["get", "list", "watch"]
+```
+
+This configuration advice SHALL NOT indicate a requirement for Kubernetes RBAC
+support.
+
+<!-- TODO: define aggregated API roles
+
+Ref:
+- https://github.com/knative/specs/blob/main/specs/eventing/channel.md#aggregated-channelable-manipulator-clusterrole
+- https://github.com/knative/specs/blob/main/specs/eventing/channel.md#aggregated-addressable-resolver-clusterrole
+-
+-->
+
+## Resource Overview
+
+The Knative Eventing API provides a set of primitives to support both
+point-to-point communication channels (`messaging.knative.dev`) and
+content-based event routing (`eventing.knative.dev`). This specification
+describes API interfaces of Knative Eventing resources as well as the supported
+[event routing](./data-plane.md) logic and configuration settings.
+
+At the moment, the Knative Eventing specification does not contemplate any
+non-Kubernetes-backed implementations, and therefore does not specifically
+define the mapping of kubernetes verbs (read, watch, patch, etc) to developer
+roles. See the [Overview documentation](./overview.md) for general definitions
+of the different API objects.
+
+## Error Signalling
+
+<!-- copied from ../serving/knative-api-specification-1.0.md#error-signalling -->
+
+The Knative API uses the
+[Kubernetes Conditions convention](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties)
+to communicate errors and problems to the user. Each user-visible resource
+described in Resource Overview MUST have a `conditions` field in `status`, which
+must be a list of `Condition` objects of the following form (note that the
+actual API object types may be named `FooCondition` to allow better code
```

Comment:
s/may/might/
I don't grok this sentence - what does naming something FooCondition mean/do ?